virtcontainers/network.go

```diff
@@ -1267,6 +1278,47 @@ func bridgeNetworkPair(endpoint Endpoint, numCPUs uint32, disableVhostNet bool)
 	return nil
 }
 
+func tapNetwork(endpoint Endpoint, numCPUs uint32, disableVhostNet bool) error {
+	netHandle, err := netlink.NewHandle()
+	if err != nil {
+		return err
+	}
+	defer netHandle.Delete()
+
+	netPair := endpoint.NetworkPair()
```

Comment:
Yes, it is much clear to use another type. I will update this.

Reply:
@caoruidong I agree with @amshinde here. It'd be better to create a new type, this would make things much clearer.

The argument of not being able to reuse the code defined in `VirtualEndpoint` does not sound valid to me, as we should be able to factorize (given the fact that we make it modular enough of course).

I think some refactoring is really necessary here as it's not appropriate to use VirtualEndpoint if your interface is not a `veth`.
But this should go further and as discussed with @amshinde, the type `VirtualEndpoint` should only be in charge of `veth`, which would make some code cleaner (we should not fallback onto `VirtualEndpoint` by default).

Reply:
Fine. I'll rewrite this part.

Reply:
@sboeuf I have taken care of the default fallback case so that VirtualEndpoint will only be created in case of veth in my PR for network refactor:
https://github.com/kata-containers/runtime/pull/811/commits/ab15498bdfb1c24fc1e15bd4dabd919adc19f6b3

Reply:
Nice, thanks!